sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoSchemaTest.java

```diff
@@ -442,4 +463,188 @@ public void process(@Element InferredPojo2 pojo, OutputReceiver<String> r) {
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(integerField, stringField, ints);
+    }
+  }
+
+  @Test
+  @Category({ValidatesRunner.class, UsesSchema.class})
+  public void testSchemaFieldSelectionUnboxing() {
+    List<PojoForExtraction> pojoList =
+        Lists.newArrayList(
+            new PojoForExtraction("a", 1, Lists.newArrayList(1, 2)),
+            new PojoForExtraction("b", 2, Lists.newArrayList(2, 3)),
+            new PojoForExtraction("c", 3, Lists.newArrayList(3, 4)));
+
+    PCollection<String> output =
+        pipeline
+            .apply(Create.of(pojoList))
+            .apply(
+                ParDo.of(
+                    new DoFn<PojoForExtraction, String>() {
+                      // Read the list twice as two equivalent types to ensure that Beam properly
+                      // converts.
+                      @ProcessElement
+                      public void process(
+                          @FieldAccess("stringField") String stringField,
+                          @FieldAccess("integerField") Integer integerField,
+                          @FieldAccess("ints") Integer[] intArray,
+                          @FieldAccess("ints") List<Integer> intList,
+                          OutputReceiver<String> r) {
+
+                        r.output(
+                            stringField
+                                + ":"
+                                + integerField
+                                + ":"
+                                + Arrays.toString(intArray)
+                                + ":"
+                                + intList.toString());
+                      }
+                    }));
+    PAssert.that(output)
+        .containsInAnyOrder("a:1:[1, 2]:[1, 2]", "b:2:[2, 3]:[2, 3]", "c:3:[3, 4]:[3, 4]");
+    pipeline.run();
+  }
+
+  @Test
+  @Category({ValidatesRunner.class, UsesSchema.class})
+  public void testSchemaFieldDescriptorSelectionUnboxing() {
+    List<PojoForExtraction> pojoList =
+        Lists.newArrayList(
+            new PojoForExtraction("a", 1, Lists.newArrayList(1, 2)),
+            new PojoForExtraction("b", 2, Lists.newArrayList(2, 3)),
+            new PojoForExtraction("c", 3, Lists.newArrayList(3, 4)));
+
+    PCollection<String> output =
+        pipeline
+            .apply(Create.of(pojoList))
+            .apply(
+                ParDo.of(
+                    new DoFn<PojoForExtraction, String>() {
+                      @FieldAccess("stringSelector")
+                      final FieldAccessDescriptor stringSelector =
+                          FieldAccessDescriptor.withFieldNames("stringField");
+
+                      @FieldAccess("intSelector")
+                      final FieldAccessDescriptor intSelector =
+                          FieldAccessDescriptor.withFieldNames("integerField");
+
+                      @FieldAccess("intsSelector")
+                      final FieldAccessDescriptor stringsSelector =
```

Comment:
don't understand

Reply:
I was referring to the variable name - it looks like a copy-paste error since the `FieldAccess` string, `intsSelector`, doesn't match the variable name, `stringSelector`

Reply:
Fixed variable name